src/Printers/Printer.ts

```diff
@@ -0,0 +1,374 @@
+        // for reasons I can't figure out some printers will refuse to return
+        // a valid config. Mostly EPL models.
+        // Give it 3 chances before we give up.
+        let retryLimit = 3;
+        do {
+            retryLimit--;
+
+            // Start listening for the return from the
+            const listenEpl = this.listenForData();
+
+            // Config isn't set up yet, send command directly without the send command.
+            await this.printerChannel.sendCommands(compiled.commandBufferRaw);
+            const rawResult = await listenEpl;
+            config = cmdSet.parseConfigurationResponse(
+                rawResult,
+                this._printerCommunicationOptions
+            );
+        } while (!config.valid && retryLimit > 0);
+
+        this.logIfDebug(`Config result is ${config.valid ? 'valid' : 'not valid.'}`);
+
+        return config;
+    }
+
+    /** Wait for the next line of data sent from the printer, or null if nothing is received. */
+    private async nextLine(timeoutMs: number): Promise<string | void> {
+        // If we have a cached line return that.
+        if (this.nextLineCache) {
+            const line = this.nextLineCache;
+            this.nextLineCache = null;
+            return line;
+        }
+
+        let timedOut = false;
+        let reader: ReadableStreamDefaultReader<string>;
+        const nextLinePromise = (async () => {
+            reader = this.printerChannel.streamFromPrinter.getReader();
+            const { value, done } = await reader.read();
+            reader.releaseLock();
+
+            if (done) {
+                return;
+            }
+
+            if (timedOut) {
+                this.nextLineCache = value;
+                return;
+            }
+
+            return value;
+        })();
+
+        const timeoutPromise = new Promise<void>((resolve) => {
+            setTimeout(() => {
+                timedOut = true;
+                reader.releaseLock();
+                resolve();
+            }, timeoutMs);
+        });
+
+        return Promise.race([nextLinePromise, timeoutPromise]);
+    }
+
+    /** Listen for incoming data until a timeout, assuming the source is done. */
+    private async listenForData(timeoutMs = 300) {
+        let aggregate = '';
+        for (;;) {
+            const line = await this.nextLine(timeoutMs);
+            if (line === undefined) {
+                this.logIfDebug(
+                    'Received',
+                    aggregate.length,
+                    'long message from printer:\n',
+                    aggregate
+                );
+                return aggregate;
+            }
+            aggregate += line + '\n';
+        }
+    }
+
+    private logIfDebug(...obj: unknown[]) {
+        if (this._printerCommunicationOptions.debug) {
+            console.debug(...obj);
+        }
+    }
+}
+
+/** Collection of handy documents ready to go. */
+export class ReadyToPrintDocuments {
```

Comment:
One way I know when a class should be in a new file or not is if the file itself is more than 200 lines.